deployment/cdk/opensearch-service-migration/lib/network-stack.ts

```diff
@@ -63,9 +77,8 @@ export class NetworkStack extends Stack {
 
         // Retrieve original deployment VPC for addon deployments
         if (props.addOnMigrationDeployId) {
-            const vpcId = StringParameter.valueForStringParameter(this, `/migration/${props.stage}/${props.defaultDeployId}/vpcId`)
             this.vpc = Vpc.fromLookup(this, 'domainVPC', {
-                vpcId: vpcId,
+                vpcId: props.vpcId,
```

Comment:
I've updated this (and improved it), I'm not convinced this was working before as the SSM call was using defaultDeployId `const vpcId = StringParameter.valueForStringParameter(this,`/migration/${props.stage}/${props.defaultDeployId}/vpcId`)`

I've made this now use the "v2 typed apis" and the valueFromLookup call so that we don't get the VPC needs a token issues, also passing in `addOnMigrationDeployId` to the SSM lookup

```
const vpcId = StringParameter.valueFromLookup(this,
                getMigrationStringParameterName({
                    stage: props.stage,
                    defaultDeployId: props.addOnMigrationDeployId,
                    parameter: MigrationSSMParameter.VPC_ID
                })
            )
```

Reply:
I think we may be on different pages here. The original intent here was to use the defaultDeployId in a sense to say, I am an additional deployment and I want to use this resource (in this case the existing vpcId) from the default main deployment for my VPC. I wouldn't expect an addOnMigration deployment to create a VPC but likely just a Replayer and maybe a Domain. valueFromLookup shouldn't be necessary here as we expect the SSM parameter to already exist, or maybe I missing something